gobblin-runtime/src/main/java/org/apache/gobblin/runtime/template/InheritingJobTemplate.java

```diff
@@ -75,23 +79,48 @@ private void resolveTemplates(Map<URI, JobTemplate> loadedTemplates) throws Spec
     if (this.resolved) {
       return;
     }
-    this.superTemplates = Lists.newArrayList();
-    for (URI uri : this.superTemplateUris) {
-      if (!loadedTemplates.containsKey(uri)) {
-        JobTemplate newTemplate = this.catalog.getTemplate(uri);
-        loadedTemplates.put(uri, newTemplate);
-        if (newTemplate instanceof InheritingJobTemplate) {
-          ((InheritingJobTemplate) newTemplate).resolveTemplates(loadedTemplates);
+
+    if (this.superTemplateUris.size() > 0) {
```

Comment:
Maybe add some javadoc for `resolveTemplates` method since this logic is a bit confusing

Reply:
Added